kdc/bx509d.c

```diff
@@ -423,18 +416,20 @@ bad_reqv(struct MHD_Connection *connection,
     char *formatted = NULL;
     char *msg = NULL;
 
-    get_krb5_context(&context);
-
+    _kdc_audit_addkv((kdc_request_t)r, 0, "http-status-code", "%d",
+                     http_status_code);
+    (void) gettimeofday(&r->tv_end, NULL);
     if (code == ENOMEM) {
-        if (context)
-            kdc_log(context, kdc_config, 4, "Out of memory");
-        return resp(connection, http_status_code, MHD_RESPMEM_PERSISTENT,
+        if (r->context)
+            kdc_log(r->context, kdc_config, 3, "Out of memory");
```

Comment:
If this is not debug logging, "our fault" errors get logged at level 1, garbage from clients at level 2, routine audit trail at level 3. So the log level should probably be 1.

Reply:
Got it.